build.sh

```diff
@@ -42,7 +42,12 @@ else
     git clone "$REPO_URL"
     cd valkey
     git checkout "$SERVER_VERSION"
-    make -j
+    make distclean
+    if [ ! -z "${ASAN_BUILD}" ]; then
+        make -j4 SANITIZER=address
```

Comment:
We don't need to specify the cores, it should be auto detected and used by the makefile
```suggestion
        make -j4 SANITIZER=address
```